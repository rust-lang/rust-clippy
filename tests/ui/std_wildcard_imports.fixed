//@aux-build:wildcard_imports_helper.rs

#![feature(test)]
#![warn(clippy::std_wildcard_imports)]

extern crate alloc;
extern crate proc_macro;
extern crate test;
extern crate wildcard_imports_helper;

use alloc::boxed::Box;
//~^ std_wildcard_imports
use core::cell::Cell;
//~^ std_wildcard_imports
use proc_macro::is_available;
//~^ std_wildcard_imports
use std::any::type_name;
//~^ std_wildcard_imports
use std::mem::{swap, align_of};
//~^ std_wildcard_imports
use test::bench::black_box;
//~^ std_wildcard_imports

use crate::fn_mod::*;

use wildcard_imports_helper::*;

use std::io::prelude::*;
use wildcard_imports_helper::extern_prelude::v1::*;
use wildcard_imports_helper::prelude::v1::*;

mod fn_mod {
    pub fn foo() {}
}

mod super_imports {
    use super::*;

    fn test_super() {
        fn_mod::foo();
    }
}

fn main() {
    foo();

    let _ = Box::new(()); // imported from alloc::boxed module
    let _ = Cell::new(()); // imported from core::cell module
    let _ = is_available(); // imported from proc_macro crate
    let _ = type_name::<i32>(); // imported from std::any module
    let _ = align_of::<i32>(); // imported from std::mem module
    black_box(()); // imported from test crate
}
