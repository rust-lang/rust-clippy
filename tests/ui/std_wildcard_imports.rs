//@aux-build:wildcard_imports_helper.rs

#![feature(test)]
#![warn(clippy::std_wildcard_imports)]

extern crate alloc;
extern crate proc_macro;
extern crate test;
extern crate wildcard_imports_helper;

use alloc::boxed::*;
//~^ std_wildcard_imports
use core::cell::*;
//~^ std_wildcard_imports
use proc_macro::*;
//~^ std_wildcard_imports
use std::any::*;
//~^ std_wildcard_imports
use std::mem::{swap, *};
//~^ std_wildcard_imports
use test::bench::*;
//~^ std_wildcard_imports

use crate::fn_mod::*;

use wildcard_imports_helper::*;

use std::io::prelude::*;
use wildcard_imports_helper::extern_prelude::v1::*;
use wildcard_imports_helper::prelude::v1::*;

mod fn_mod {
    pub fn foo() {}
}

mod super_imports {
    use super::*;

    fn test_super() {
        fn_mod::foo();
    }
}

fn main() {
    foo();

    let _ = Box::new(()); // imported from alloc::boxed module
    let _ = Cell::new(()); // imported from core::cell module
    let _ = is_available(); // imported from proc_macro crate
    let _ = type_name::<i32>(); // imported from std::any module
    let _ = align_of::<i32>(); // imported from std::mem module
    black_box(()); // imported from test crate
}
