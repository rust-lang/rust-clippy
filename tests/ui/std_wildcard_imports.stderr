error: usage of wildcard import from `alloc`
  --> tests/ui/std_wildcard_imports.rs:11:5
   |
LL | use alloc::boxed::*;
   |     ^^^^^^^^^^^^^^^ help: try: `alloc::boxed::Box`
   |
   = note: `-D clippy::std-wildcard-imports` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::std_wildcard_imports)]`

error: usage of wildcard import from `core`
  --> tests/ui/std_wildcard_imports.rs:13:5
   |
LL | use core::cell::*;
   |     ^^^^^^^^^^^^^ help: try: `core::cell::Cell`

error: usage of wildcard import from `proc_macro`
  --> tests/ui/std_wildcard_imports.rs:15:5
   |
LL | use proc_macro::*;
   |     ^^^^^^^^^^^^^ help: try: `proc_macro::is_available`

error: usage of wildcard import from `std`
  --> tests/ui/std_wildcard_imports.rs:17:5
   |
LL | use std::any::*;
   |     ^^^^^^^^^^^ help: try: `std::any::type_name`

error: usage of wildcard import from `std`
  --> tests/ui/std_wildcard_imports.rs:19:22
   |
LL | use std::mem::{swap, *};
   |                      ^ help: try: `align_of`

error: usage of wildcard import from `test`
  --> tests/ui/std_wildcard_imports.rs:21:5
   |
LL | use test::bench::*;
   |     ^^^^^^^^^^^^^^ help: try: `test::bench::black_box`

error: aborting due to 6 previous errors

